tests/main/interfaces-opengl-nvidia/task.yaml

```
@@ -90,35 +89,46 @@ restore: |
     rm -rf /usr/lib32/nvidia-123
 
 execute: |
-    "$TESTSTOOLS"/snaps-state install-local test-snapd-policy-app-consumer
+    "$TESTSTOOLS"/snaps-state install-local gl-core16
 
     echo "When the interface is connected"
-    snap connect test-snapd-policy-app-consumer:opengl core:opengl
+    snap connect gl-core16:opengl core:opengl
 
     echo "App can access nvidia library files"
-    expected="canary-legacy"
-    if os.query is-bionic; then
-        expected="canary-triplet"
+    expected="canary-triplet"
+    if os.query is-xenial; then
+        expected="canary-legacy"
     fi
     files="libGLX.so.0.0.1 libGLX_nvidia.so.0.0.1 libnvidia-glcore.so.$NV_VERSION tls/libnvidia-tls.so.$NV_VERSION libnvidia-tls.so.$NV_VERSION vdpau/libvdpau_nvidia.so.$NV_VERSION"
     for f in $files; do
-       snap run test-snapd-policy-app-consumer.opengl -c "cat /var/lib/snapd/lib/gl/$f" | MATCH "$expected"
+       gl-core16 cat "/var/lib/snapd/lib/gl/$f" | MATCH "$expected"
     done
 
     if os.query is-pc-amd64; then
-        expected32="canary-32-legacy"
-        if os.query is-bionic; then
-            expected32="canary-32-triplet"
+        expected32="canary-32-triplet"
+        if os.query is-xenial; then
+            expected32="canary-32-legacy"
         fi
         for f in $files; do
-            snap run test-snapd-policy-app-consumer.opengl -c "cat /var/lib/snapd/lib/gl32/$f" | MATCH "$expected32"
+            gl-core16 cat "/var/lib/snapd/lib/gl32/$f" | MATCH "$expected32"
         done
     fi
 
     echo "And vulkan ICD file"
-    snap run test-snapd-policy-app-consumer.opengl -c "cat /var/lib/snapd/lib/vulkan/icd.d/nvidia_icd.json" | MATCH canary-vulkan
+    gl-core16 cat /var/lib/snapd/lib/vulkan/icd.d/nvidia_icd.json | MATCH canary-vulkan
 
-    if os.query is-bionic; then
+    if ! os.query is-xenial; then
         echo "And GLVND EGL vendor file"
-        snap run test-snapd-policy-app-consumer.opengl -c "cat /var/lib/snapd/lib/glvnd/egl_vendor.d/10_nvidia.json" | MATCH canary-egl
+        gl-core16 cat /var/lib/snapd/lib/glvnd/egl_vendor.d/10_nvidia.json | MATCH canary-egl
+    fi
+
+    echo "For host systems using glvnd, the glvnd libraries are not exposed to snaps using newer bases"
+    "$TESTSTOOLS"/snaps-state install-local gl-core20
+    snap connect gl-core20:opengl core:opengl
+
+    if ! os.query is-xenial; then
+        not gl-core20 test -f /var/lib/snapd/lib/gl/libGLX.so.0.1
+        if os.query is-pc-amd64; then
+            not gl-core16 cat /var/lib/snapd/lib/gl32/libGLX.so.0.1
+        fi
```

The libs have 0.0.1 version suffix. Also it's be good to verify that libGLX_nvidia is available.
```suggestion
        not gl-core20 test -f /var/lib/snapd/lib/gl/libGLX.so.0.0.1
        gl-core20 cat /var/lib/snapd/lib/gl/libGLX_nvidia.so.0.0.1 | MATCH canary-triplet
        if os.query is-pc-amd64; then
            not gl-core16 cat /var/lib/snapd/lib/gl32/libGLX.so.0.0.1
            gl-core20 cat /var/lib/snapd/lib/gl32/libGLX_nvidia.so.0.0.1 | MATCH canary-32-triplet
        fi
```